contracts/VeJoeStaking.sol

```diff
@@ -46,35 +66,62 @@ contract VeJoeStaking is Initializable, OwnableUpgradeable {
+    uint256 public speedUpVeJoePerSharePerSec;
+
+    /// @notice Precision of `veJoePerSharePerSec`
+    uint256 public VEJOE_PER_SHARE_PER_SEC_PRECISION;
+
+    /// @notice Percentage of user's current staked JOE user has to deposit in order to start
+    /// receiving speed up benefits, in parts per 100.
+    /// @dev Specifically, user has to deposit at least `speedUpThreshold/100 * userStakedJoe` JOE.
+    /// The only exception is the user will also receive speed up benefits if it's their first
+    /// time staking.
+    uint256 public speedUpThreshold;
+
+    /// @notice The length of time a user receives speed up benefits
+    uint256 public speedUpDuration;
 
     mapping(address => UserInfo) public userInfos;
 
     event Claim(address indexed user, uint256 amount);
     event Deposit(address indexed user, uint256 amount);
-    event UpdateVeJoePerSec(address indexed user, uint256 veJoePerSec);
     event UpdateMaxCap(address indexed user, uint256 maxCap);
     event UpdateRewardVars(uint256 lastRewardTimestamp, uint256 accVeJoePerShare);
+    event UpdateSpeedUpThreshold(address indexed user, uint256 speedUpThreshold);
+    event UpdateVeJoePerSharePerSec(address indexed user, uint256 veJoePerSharePerSec);
     event Withdraw(address indexed user, uint256 amount);
 
     /// @notice Initialize with needed parameters
     /// @param _joe Address of the JOE token contract
     /// @param _veJoe Address of the veJOE token contract
-    /// @param _veJoePerSec veJOE per sec per JOE staked, scaled to `VEJOE_PER_SEC_PRECISION`
-    /// @param _maxCap the maximum amount of veJOE received per JOE staked
+    /// @param _veJoePerSharePerSec veJOE per sec per JOE staked, scaled to `VEJOE_PER_SHARE_PER_SEC_PRECISION`
+    /// @param _speedUpVeJoePerSharePerSec Similar to `_veJoePerSharePerSec` but for speed up
+    /// @param _speedUpThreshold Percentage of total staked JOE user has to deposit receive speed up
+    /// @param _speedUpDuration Length of time a user receives speed up benefits
+    /// @param _maxCap The maximum amount of veJOE received per JOE staked
     function initialize(
         IERC20Upgradeable _joe,
         VeJoeToken _veJoe,
-        uint256 _veJoePerSec,
+        uint256 _veJoePerSharePerSec,
+        uint256 _speedUpVeJoePerSharePerSec,
+        uint256 _speedUpThreshold,
+        uint256 _speedUpDuration,
         uint256 _maxCap
     ) public initializer {
         require(address(_joe) != address(0), "VeJoeStaking: unexpected zero address for _joe");
         require(address(_veJoe) != address(0), "VeJoeStaking: unexpected zero address for _veJoe");
 
+        require(
+            _speedUpThreshold != 0 && _speedUpThreshold <= 100,
+            "VeJoeStaking: expected _speedUpThreshold to be > 0 and <= 100"
+        );
+
+        require(_speedUpDuration <= 365 days, "VeJoeStaking: expected _speedUpDuration to be <= 365 days");
```

Comment:
Is it intended for 0 to be a valid value here? I assume if we wanted to disable sped up generation we set to 0.

Reply:
yeah I don't feel too strongly about this. I wanted to leave it open as an option in case we did want to set it to 0 in the future